controller/src/main/java/org/jboss/as/controller/persistence/FilePersistenceUtils.java

```diff
@@ -77,9 +77,13 @@ static void copyFile(final File file, final File backup) throws IOException {
         Files.copy(file.toPath(), backup.toPath(), StandardCopyOption.COPY_ATTRIBUTES, StandardCopyOption.REPLACE_EXISTING);
     }
 
-    static void rename(File file, File to) throws IOException {
-        if (!file.renameTo(to) && file.exists()) {
-            copyFile(file, to);
+    static boolean rename(Path from, Path to) {
+        try {
+            Files.move(from, to);
+            return true;
+        } catch (IOException e) {
+            ControllerLogger.ROOT_LOGGER.cannotRename(e, from, to);
+            return false;
```

Comment:
Note: this method wasn't used at all in current code so there's no need to worry about any behavior change from this part of the PR. It's just whether the logic is correct.